giskard/scanner/scanner.py

```diff
@@ -183,6 +148,64 @@ def _postprocess(self, issues: Sequence[Issue]) -> Sequence[Issue]:
 
         return issues
 
+    def _validate_model_and_dataset(self, model, dataset):
+        if not isinstance(model, BaseModel):
+            raise ValueError(
+                "The model object you provided is not valid. Please wrap it with the `giskard.Model` class. "
+                "See the instructions here: https://docs.giskard.ai/en/latest/guides/wrap_model/index.html"
+            )
+
+        if dataset is not None and not isinstance(dataset, Dataset):
+            raise ValueError(
+                "The dataset object you provided is not valid. Please wrap your dataframe with `giskard.Dataset`. "
+                "You can follow the docs here: https://docs.giskard.ai/en/latest/guides/wrap_dataset/index.html"
+            )
+
+        # Some extra checks
+        if model.is_text_generation:
+            if not model.meta.name or not model.meta.description:
+                raise ValueError(
+                    "LLM models must have a name and a description. Please make sure the `name` and `description` "
+                    "parameters of your model are not empty. We will use them to configure our detectors."
+                )
+            if not model.meta.feature_names:
+                raise ValueError(
+                    "LLM models must specify their input variables to be analyzed. "
+                    "Please make sure to set the `feature_names` parameter when wrapping your model."
+                )
+
+        model, dataset = self._prepare_model_dataset(model, dataset)
+
+        if not model.is_text_generation:
+            time_start = perf_counter()
+            validate_model(model=model, validate_ds=dataset)
+            model_validation_time = perf_counter() - time_start
+        else:
+            model_validation_time = None
+
+        if not dataset.df.index.is_unique:
+            warning(
+                "You dataframe has duplicate indexes, which is currently not supported. "
+                "We have to reset the dataframe index to avoid issues."
+            )
+            dataset = Dataset(
+                df=dataset.df.reset_index(drop=True),
+                name=dataset.name,
+                target=dataset.target,
+                column_types=dataset.column_types,
+            )
```

Comment:
I think that is generally problematic for us, so i would move it to `dataset_validation`

Reply:
out of scope for this PR